### ISSUE TITLE
Improve docs for t.w.h.Request.addCookie

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#9479]
|[<img alt="evilham's avatar" src="https://avatars.githubusercontent.com/u/4446607?s=50" width="50" height="50">](https://github.com/evilham)| @evilham reported|
|-|-|
|Trac ID|trac#9479|
|Type|enhancement|
|Created|2018-07-04 11:58:49Z|
|Branch|https://github.com/twisted/twisted/tree/9480-evilham-web-Request-addCookie-sameSite|

This is related to [#9387](https://github.com/twisted/twisted/issues/9387)

API Docs don't list the possible values for the `samesite` parameter, they also don't mention that if something invalid gets passed, a `ValueError` is raised.

Maybe, as in the trac issue, a link to the Draft RFC would be useful for library users.

<details><summary>Searchable metadata</summary>

```
trac-id__9479 9479
type__enhancement enhancement
reporter__evilham evilham
priority__low low
milestone__None None
branch__9480_evilham_web_Request_addCookie_sameSite 9480-evilham-web-Request-addCookie-sameSite
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation_easy documentation easy
time__1530705529582408 1530705529582408
changetime__1530777600383815 1530777600383815
version__None None
owner__None None
cc__alex
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="evilham's avatar" src="https://avatars.githubusercontent.com/u/4446607?s=50" width="50" height="50">](https://github.com/evilham)<a name="note_3"></a>|@evilham commented|
|-|-|

https://github.com/twisted/twisted/pull/1036

**Comment by twisted-trac** [comment]
|[<img alt="evilham's avatar" src="https://avatars.githubusercontent.com/u/4446607?s=50" width="50" height="50">](https://github.com/evilham)<a name="note_4"></a>|@evilham commented|
|-|-|

Since [#9480](https://github.com/twisted/twisted/issues/9480) blocks this and they are quite related, fixed together in the same PR.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

See PR.

**Comment by twisted-trac** [comment]
|[<img alt="evilham's avatar" src="https://avatars.githubusercontent.com/u/4446607?s=50" width="50" height="50">](https://github.com/evilham)<a name="note_6"></a>|@evilham set status to `closed`|
|-|-|

Fixed and merged with [#9480](https://github.com/twisted/twisted/issues/9480)
